### PR TITLE
tests: lib: move heap to new ztest API

### PR DESCRIPTION
**Pull request by 1xlibiao**
Move test lib/heap to use new ztest API.

Signed-off-by: li biao <biao1x.li@intel.com>

### CODE REVIEW

**Review by stephanosio — DISMISSED** [review]
Please fix the typo in the commit title:
```
itests: lib: move heap to new ztest API 
```

**Comment by 1xlibiao** [comment]
> Please fix the typo in the commit title:
> 
> ```
> itests: lib: move heap to new ztest API 
> ```

OK,thanks for helping me correct,I will modify it as soon.

File: tests/lib/heap/src/main.c [hunk 1]
```diff
@@ -216,7 +216,7 @@ static void test_fragmentation(void)
  * exhaustively with good performance, so the relative operation count
  * and fragmentation is going to be lower.
  */
-static void test_big_heap(void)
+ZTEST(lib_heap_test, test_big_heap)
```
Comment (nashif) [inline tests/lib/heap/src/main.c:219]: no need for _test in the suite name.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/lib/heap/src/main.c (hunk 1 above)
Comment (1xlibiao) [reply tests/lib/heap/src/main.c:219]: ok,I will revise it as soon

**Review by chen-png — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Review by yerabolu — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**